File: packages/dart/events/lib/src/exceptions.dart
/// Base exception for Kitbase SDK errors.
class KitbaseException implements Exception {
  final String message;

  const KitbaseException(this.message);

  @override
  String toString() => 'KitbaseException: $message';
}

/// Exception thrown when API authentication fails.
class AuthenticationException extends KitbaseException {
  const AuthenticationException([String message = 'Invalid API key'])
      : super(message);

  @override
  String toString() => 'AuthenticationException: $message';
}

/// Exception thrown when the API request fails.
class ApiException extends KitbaseException {
  final int statusCode;
  final dynamic response;

  const ApiException(
    super.message, {
    required this.statusCode,
    this.response,
  });

  @override
  String toString() => 'ApiException ($statusCode): $message';
}

/// Exception thrown when request validation fails.
class ValidationException extends KitbaseException {
  final String? field;

  const ValidationException(super.message, {this.field});

  @override
  String toString() =>
      'ValidationException${field != null ? ' ($field)' : ''}: $message';
}

/// Exception thrown when a request times out.
class TimeoutException extends KitbaseException {
  const TimeoutException([String message = 'Request timed out']) : super(message);

  @override
  String toString() => 'TimeoutException: $message';
}

File: packages/php/src/Events/Exceptions.php
<?php

declare(strict_types=1);

namespace Kitbase\Events;

use Exception;
use Throwable;

/**
 * Base exception class for Kitbase SDK errors
 */
class KitbaseException extends Exception
{
    public function __construct(string $message = '', int $code = 0, ?Throwable $previous = null)
    {
        parent::__construct($message, $code, $previous);
    }
}

/**
 * Exception thrown when API authentication fails
 */
class AuthenticationException extends KitbaseException
{
    public function __construct(string $message = 'Invalid API key', int $code = 401, ?Throwable $previous = null)
    {
        parent::__construct($message, $code, $previous);
    }
}

/**
 * Exception thrown when the API request fails
 */
class ApiException extends KitbaseException
{
    public readonly int $statusCode;
    public readonly mixed $response;

    public function __construct(
        string $message,
        int $statusCode,
        mixed $response = null,
        ?Throwable $previous = null
    ) {
        parent::__construct($message, $statusCode, $previous);
        $this->statusCode = $statusCode;
        $this->response = $response;
    }
}

/**
 * Exception thrown when request validation fails
 */
class ValidationException extends KitbaseException
{
    public readonly ?string $field;

    public function __construct(string $message, ?string $field = null, ?Throwable $previous = null)
    {
        parent::__construct($message, 0, $previous);
        $this->field = $field;
    }
}

/**
 * Exception thrown when a request times out
 */
class TimeoutException extends KitbaseException
{
    public function __construct(string $message = 'Request timed out', ?Throwable $previous = null)
    {
        parent::__construct($message, 0, $previous);
    }
}








